pkg/controller/status/status.go

```diff
@@ -106,6 +107,9 @@ func (m *statusManager) Run() {
 		// Loop forever, periodically checking dependent objects for their state.
 		for {
 			if !m.syncState() {
+				if !m.enabled {
```

Comment:
Looks like everywhere else we access `m.enabled` we're using a mutex to prevent concurrent access. I think we should do that here as well.

Reply:
We are using mutex while writing into `m.enabled`, looks like we don't use it for read https://github.com/tigera/operator/blob/master/pkg/controller/status/status.go#L455

Reply:
@Suraiya-Hameed if you look at all the places that we call `set(` you will see that the mutex is already held by the caller.